tests/framework_sanic/conftest.py

```diff
@@ -123,6 +126,15 @@ def request(app, method, url, headers=None):
     if loop is None:
         loop = asyncio.new_event_loop()
 
+    if not getattr(app.router, "finalized", True):
+        # Handle startup if the router hasn't been finalized.
+        # Older versions don't have this requirement or variable so
+        # the default should be True.
+        if hasattr(app, "_startup"):
+            loop.run_until_complete(app._startup())
+        else:
+            app.router.finalize()
```

Comment:
Does the call to `finalize()` need to be wrapped in a try/ except to account for a potential `FinalizationError`? That error would only be raised if the router has already been finalized (which shouldn't apply in this context) or if there are no routes picked up to finalize.

Reply:
This is just for testing setup so if the router fails to init we'll want it to error out.